backend/frontend/views.py

```diff
@@ -426,6 +449,64 @@ def progress_details(
     )
 
 
+def check_user_can_add_review(logged_in_user):
+    request = get_current_request()
+    content_type = request.resolver_match.kwargs.get("content_type")
+    id = request.resolver_match.kwargs.get("id")
+
+    if content_type == "project":
+        project = get_object_or_404(RecruitProject, id=id)
+        return project.request_user_can_add_review(logged_in_user)
+
+    return False
+
+@user_passes_test_or_forbidden(check_user_can_add_review)
+def action_add_review(request, content_type, id):
+    if request.method == "POST":
```

Comment:
This function has a lot of nesting, which makes reading it a bit hard to follow.

I don't think checking the method is as seen with the other action related methods and the fact that this will always be a POST.